### ISSUE TITLE
[Feature] dbt seeds - STRING as default for the entire project

### ISSUE THREAD

**Issue by a51001k** [dbt-labs/dbt-core#10985]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

dbt seeds seemingly automatically decides data types before inserting data into for example Snowflake.

In our use-case it would be helpful if it was possible to set `string `as default for the entire dbt project.

That way our pipelines would be more predictable, and we can ourselves change `string` into `number`/`boolean`/`date` etc. in a staging-layer.

### Describe alternatives you've considered

Hardcoding every single column into string. This is a pain.

### Who will this benefit?

Everyone who wants to have predictable dbt seeds and treat seeds as an ingest.

**Comment by kmarq** [comment]
Tried to override this behavior, but it had numerous challenges to do. I'd definitely appreciate this as an option. It allows loading seed files without defining each and every column in a yaml if all you want is for it to load as strings. 
Even being able to default to String on a per seed file basis that'd be a big improvement.

**Comment by steveharshman-wk** [comment]
I would love to see this feature as I have an example where the seed data is often changing and includes columns that I cannot know beforehand.

**Comment by graciegoheen** [comment]
Hi @a51001k - you're correct. Today, dbt will infer the datatype for each column based on the data in your CSV.

> That way our pipelines would be more predictable

You can override that by explicitly setting the data types using the `column_types:` config:
```
seeds:
  jaffle_shop:
    warehouse_locations:
      +column_types:
        zipcode: varchar(5)
```

I understand this is cumbersome for you to set for every single seed column, but you should only need to set it once for each seed you have.

> I have an example where the seed data is often changing and includes columns that I cannot know beforehand

Can you say more about this use case? We say in our [docs](https://docs.getdbt.com/docs/build/seeds?version=1.12):

> Seeds are best suited to static data which changes infrequently.

So seeds might not be the right solution for you.

**Comment by steveharshman-wk** [comment]
@graciegoheen I did decide on a different solution for this use case. The seed data I have is a csv which contains a number of defined columns, but can contain additional columns which will be ignored on ingestion. I don't have control over these additional columns so I cannot predefine the column name and data types. This led to potential ingestion issues with the dbt seed solution.

**Comment by graciegoheen** [comment]
Makes sense. I'm going to close this issue out for now, but we can always re-open if there's renewed interest!
